bin/analysis_target_nonwindows.go

```diff
@@ -0,0 +1,26 @@
+//go:build !windows
+// +build !windows
+
+package main
+
+import (
+	config_proto "www.velocidex.com/golang/velociraptor/config/proto"
+	vql_subsystem "www.velocidex.com/golang/velociraptor/vql"
+)
+
+func applyAnalysisTarget(config_obj *config_proto.Config) error {
+	if config_obj.AnalysisTarget == "" {
+		return nil
+	}
+
+	switch config_obj.AnalysisTarget {
+	case "windows":
+		vql_subsystem.MakeNoopPlugin("wmi")
```

Comment:
We obviously need to remove more than this plugin.

I am just wondering how do we keep these in sync as new plugins are introduced? This is why I was suggesting using the ACL system instead because each plugin should declare the required ACLs and so it will naturally work that running the plugin will receive permission denied when running in this mode.

But probably this will be needed anyway to provide "stub" plugins for different OS platforms - otherwise we will receive symbol not found errors on Linux for wmi for example.

Reply:
Yes, this definitely needs to be done to more than the wmi plugin. I picked the wmi plugin only because I knew it did not work on linux to see how my idea of replacing plugins feels like. And to be honest: although it works, I do not really like the conditional compilation either. But I did not know of a more appropriate method of doing certain things based on the host OS, so I went for the "ugly" way. However, I will have a thorough look into the ACL system and see how it works out if we use ACLs instead of conditional compilation.